src/Chapter3.hs

```diff
@@ -971,6 +1094,31 @@ implement the following functions:
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
 
+data DayOfWeek
+  = Monday
+  | Tuesday
+  | Wednesday
+  | Thursday
+  | Friday
+  | Saturday
+  | Sunday
+  deriving (Show, Read, Enum, Bounded, Eq, Ord)
+
+isWeekend :: DayOfWeek -> Bool
+isWeekend day
+  | day == Saturday || day == Sunday = True
+  | otherwise = False
+
+nextDay :: DayOfWeek -> DayOfWeek
+nextDay day
+  | day == maxBound = minBound
```

Comment:
:+1: This is also a bit more maintainable. If you decide, for example, to have the "American" system of week days in the data type constructors, you won't need to change this function at all! 🙂